pkg/catalog/catalog.go

```diff
@@ -2651,7 +2651,7 @@ func (c *Catalog) PrepareGCUncommitted(ctx context.Context, repositoryID string,
 }
 
 // CopyEntry copy entry information by using the block adapter to make a copy of the data to a new physical address.
-func (c *Catalog) CopyEntry(ctx context.Context, srcRepository, srcRef, srcPath, destRepository, destBranch, destPath string, opts ...graveler.SetOptionsFunc) (*DBEntry, error) {
+func (c *Catalog) CopyEntry(ctx context.Context, srcRepository, srcRef, srcPath, destRepository, destBranch, destPath string, copySrcMetadata bool, metadata Metadata, opts ...graveler.SetOptionsFunc) (*DBEntry, error) {
```

Comment:
Why do we need both?
If we pass metadata - override

Reply:
Also can we document it?

Reply:
https://github.com/treeverse/lakeFS/pull/8117#discussion_r1741504142

Reply:
In short, CopyObject from lakeFS API will pass `nil` metadata and also shouldn't copy the source metadata

Reply:
You can differentiate nil and empty slice to avoid that

Reply:
Do you think it's better? Feel so fragile and easy to make mistakes.

Reply:
Personally I feel like it's cleaner and less redundant.
I would however add simple tests to cover these 3 scenarios

Reply:
Not blocking on it though